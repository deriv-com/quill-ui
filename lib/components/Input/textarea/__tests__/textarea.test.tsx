import { act, fireEvent, render, screen } from "@testing-library/react";
import TextArea from "..";

describe("TextArea Component", () => {
    it("renders with default props", () => {
        const { container } = render(<TextArea />);
        expect(container.querySelector("textarea")).toBeInTheDocument();
    });

    it("handles onChange correctly", () => {
        const handleChange = jest.fn();
        const { getByPlaceholderText } = render(
            <TextArea placeholder="Test Placeholder" onChange={handleChange} />,
        );
        const textarea = getByPlaceholderText("Test Placeholder");
        act(() => {
            fireEvent.change(textarea, { target: { value: "Test" } });
        });
        expect(handleChange).toHaveBeenCalled();
        expect(textarea).toMatchSnapshot();
    });

    it("displays character counter when show_counter is true", () => {
        const { getByText } = render(
            <TextArea maxLength={50} show_counter={true} value="Test" />,
        );
        expect(getByText("4/50")).toBeInTheDocument();
    });

    it("renders with the correct size", () => {
        const { container } = render(<TextArea size="sm" />);
        expect(screen.getByRole("textbox").parentElement).toHaveClass(
            "quill-textarea__wrapper__size--sm",
        );
        expect(container).toMatchSnapshot();
    });

    it("displays the label when provided", () => {
        render(<TextArea label="Description" />);
        expect(screen.getByText("Description")).toBeInTheDocument();
    });

    it("updates character counter as text is entered", () => {
        render(<TextArea show_counter maxLength={100} value="Hello" />);

        expect(screen.getByText("5/100")).toBeInTheDocument();
    });

    it("shows character counter when show_counter is true and maxLength is provided", () => {
        render(<TextArea show_counter maxLength={100} />);
        expect(screen.getByText("0/100")).toBeInTheDocument();
    });
});
