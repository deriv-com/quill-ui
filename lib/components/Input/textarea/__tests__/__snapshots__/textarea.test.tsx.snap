// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea Component handles onChange correctly 1`] = `
<textarea
  class="quill-textarea"
  placeholder="Test Placeholder"
/>
`;

exports[`TextArea Component renders with the correct size 1`] = `
<div>
  <div
    class="quill-textarea__container"
  >
    <div
      class="quill-textarea__wrapper quill-textarea__wrapper__variant--outline quill-textarea__wrapper__size--sm quill-textarea__wrapper__variant--outline--neutral quill-textarea__wrapper__resize--true no-left-icon no-right-icon"
    >
      <textarea
        class="quill-textarea"
      />
    </div>
    <div
      class="quill-textarea__resizer"
    />
    <div
      class="message__container message__container--sm"
    >
      <p
        class="quill-typography quill-typography__caption-text__weight--regular__decoration--default quill-typography__color--prominent message--neutral"
      />
    </div>
  </div>
</div>
`;
